Infix sequence wildcard can't match empty sequence in 2.13 (empty multiarg infix not supported)

## reproduction steps

Scala version: 2.13.3

```scala
object Main
  extends App {

  class Unapplyer {
    def unapplySeq(seq: Seq[Int]): Option[(Int, Seq[Int])] =
      if (seq.isEmpty) None else Some((seq.head, seq.tail))
  }
  val unapplyer = new Unapplyer

  /* 
     v2.12.12:
       prints "1"
     v2.13.3:
       type mismatch;
        found   : Unit
        required: Int
           case value unapplyer () => println(value)
   */
  Seq(1) match {
    case value unapplyer () => println(value)
  }
}
```

## problem

The code above can't be compiled with Scala 2.13.

2.13:

```
$ rm -r out
$ mkdir out
$ scalac -d out $(f 'class Unapplyer { def unapplySeq(seq: Seq[Int]): Option[(Int, Seq[Int])] = if (seq.isEmpty) None else Some((seq.head, seq.tail)) }')
$ scalac -cp out -Vprint:typer $(f 'class C { val unapplyer = new Unapplyer; def m = Seq(1) match { case value unapplyer () => value } }')

/var/folders/yh/qsn4cxqj47gbhh9_prlxqy3r0000gn/T/scala-test.HDllnj05.scala:1: error: type mismatch;
 found   : Unit
 required: Int
class C { val unapplyer = new Unapplyer; def m = Seq(1) match { case value unapplyer () => value } }
                                                                           ^
[[syntax trees at end of                     typer]] // scala-test.HDllnj05.scala
package <empty> {
  class C extends scala.AnyRef {
    def <init>(): C = {
      C.super.<init>();
      ()
    };
    private[this] val unapplyer: Unapplyer = new Unapplyer();
    <stable> <accessor> def unapplyer: Unapplyer = C.this.unapplyer;
    def m: Int = scala.collection.immutable.Seq.apply[Int](1) match {
      case C.this.unapplyer.unapplySeq(<unapply-selector>) <unapply> ((value @ _), ()) => value
    }
  }
}

1 error
```

2.12:

```
$ rm -r out212
$ mkdir out212
$ scalac -212 -d out212 $(f 'class Unapplyer { def unapplySeq(seq: Seq[Int]): Option[(Int, Seq[Int])] = if (seq.isEmpty) None else Some((seq.head, seq.tail)) }')
$ scalac -212 -cp out212 -Xprint:typer $(f 'class C { val unapplyer = new Unapplyer; def m = Seq(1) match { case value unapplyer () => value } }')

[[syntax trees at end of                     typer]] // scala-test.2MnsWGH7.scala
package <empty> {
  class C extends scala.AnyRef {
    def <init>(): C = {
      C.super.<init>();
      ()
    };
    private[this] val unapplyer: Unapplyer = new Unapplyer();
    <stable> <accessor> def unapplyer: Unapplyer = C.this.unapplyer;
    def m: Int = scala.collection.Seq.apply[Int](1) match {
      case C.this.unapplyer.unapplySeq(<unapply-selector>) <unapply> ((value @ _)) => value
    }
  }
}
```

First of all I'm surprised that `def unapplySeq(seq: Seq[Int]): Option[(Int, Seq[Int])]` is even considered, as I thought (and https://docs.scala-lang.org/tour/extractor-objects.html describes) that it needs to be `Option[Seq[T]]`.

Secondly I'd never seen infix notation for `unapplySeq` - does that follows [SLS 8.1.11](https://www.scala-lang.org/files/archive/spec/2.13/08-pattern-matching.html#infix-operation-patterns)?

> ## 8.1.11 Infix Operation Patterns
> ```
>   Pattern3  ::=  SimplePattern {id [nl] SimplePattern}
> ```
> An _infix operation pattern_ `𝑝;op;𝑞` is a shorthand for the constructor or extractor pattern `op(𝑝,𝑞)`. The precedence and associativity of operators in patterns is the same as in expressions.
> 
> An infix operation pattern `𝑝;op;(𝑞1,…,𝑞𝑛)` is a shorthand for the constructor or extractor pattern `op(𝑝,𝑞1,…,𝑞𝑛)`.

So `unapplyer.unapplySeq(value, ())`, what 2.13 is doing, seems to follow the spec, right?

My feeling and vote right now is that the code shouldn't've compiled in 2.12, so it's not a bug in 2.13.

--- a/test/files/neg/t12110.check
+++ b/test/files/neg/t12110.check
@@ -1,0 +1,6 @@
+t12110.scala:20: error: type mismatch;
+ found   : Unit
+ required: Int
+    case value unapplyer () => println(value)
+               ^
+1 error

--- a/test/files/neg/t12110.scala
+++ b/test/files/neg/t12110.scala
@@ -1,0 +1,30 @@
+
+object Main extends App {
+
+  class Unapplyer {
+    def unapplySeq(seq: Seq[Int]): Option[(Int, Seq[Int])] =
+      if (seq.isEmpty) None else Some((seq.head, seq.tail))
+  }
+  val unapplyer = new Unapplyer
+
+  /*
+     v2.12.12:
+       prints "1"
+     v2.13.3:
+       type mismatch;
+        found   : Unit
+        required: Int
+           case value unapplyer () => println(value)
+   */
+  Seq(1) match {
+    case value unapplyer () => println(value)
+  }
+}
+/*
+scala 3:
+-- Error: test/files/neg/t12110.scala:20:25 ----------------------------------------------------------------------------------------------
+20 |    case value unapplyer () => println(value)
+   |                         ^^
+   |                         Values of types Unit and Int cannot be compared with == or !=
+1 error found
+*/

